README.md

```diff
@@ -4,175 +4,225 @@
-or `decrypt_access_token` respectively.
-
-Both functions take the access token, a `cipher` for the cryptographic operations and an optional `aad` (additional
-authenticated data).
+Some types used across multiple scenarios include:
 
-`decrypt_access_token` will return a result containing the decrypted
-`ClaimsSet`.
-`verify_access_token` will return an empty result which indicates that the token was successfully verified---an `Err`
+- [`Scope`](https://docs.rs/dcaf/latest/dcaf/common/scope/enum.Scope.html) (as described in
+  [section 5.8.1 of RFC 9200](https://www.rfc-editor.org/rfc/rfc9200#section-5.8.1)),
+  either as a [`TextEncodedScope`](https://docs.rs/dcaf/latest/dcaf/common/scope/struct.TextEncodedScope.html),
+  a [`BinaryEncodedScope`](https://docs.rs/dcaf/latest/dcaf/common/scope/struct.BinaryEncodedScope.html) or
+  an [`AifEncodedScope`](https://docs.rs/dcaf/latest/dcaf/common/scope/struct.AifEncodedScope.html).
+- [`ProofOfPossessionKey`](https://docs.rs/dcaf/latest/dcaf/common/cbor_values/enum.ProofOfPossessionKey.html) as
+  specified in
+  [section 3.1 of RFC 8747](https://www.rfc-editor.org/rfc/rfc8747#section-3.1).
+  For example, this will be used in the access token's `cnf` claim.
+- While not really a data type, various constants representing values used in ACE-OAuth
+  are provided in the [`constants`](https://docs.rs/dcaf/latest/dcaf/common/constants/) module.
+
+# Creating Access Tokens
+
+In order to create access tokens, you can use
+either [`encrypt_access_token`](https://docs.rs/dcaf/latest/dcaf/token/fn.encrypt_access_token.html)
+or [`sign_access_token`](https://docs.rs/dcaf/latest/dcaf/token/fn.sign_access_token.html),
+depending on whether you want the access token to be wrapped in a
+`COSE_Encrypt0` or `COSE_Sign1` structure. Support for a combination of both is planned for the
+future. In case you want to create a token intended for multiple recipients (each with their
+own key), you can
+use [`encrypt_access_token_multiple`](https://docs.rs/dcaf/latest/dcaf/token/fn.encrypt_access_token_multiple.html)
+or [`sign_access_token_multiple`](https://docs.rs/dcaf/latest/dcaf/token/fn.sign_access_token_multiple.html).
+
+Both functions take a [`ClaimsSet`](coset::cwt::ClaimsSet) containing the claims that
+shall be part of the access token, a key used to encrypt or sign the token,
+optional `aad` (additional authenticated data), un-/protected headers and a cipher (explained
+further below) identified by type parameter `T`.
+Note that if the headers you pass in set fields which the cipher wants to set as well,
+the function will fail with a `HeaderAlreadySet` error.
+The function will return a [`Result`](https://doc.rust-lang.org/stable/core/result/enum.Result.html) of the opaque
+[`ByteString`](https://docs.rs/dcaf/latest/dcaf/common/cbor_values/type.ByteString.html) containing the access token.
+
+# Verifying / decrypting Access Tokens
```

Comment:
```suggestion
# Verifying and Decrypting Access Tokens
```

Reply:
Adjusted in `lib.rs` crate-level docs, as the README is generated from it.